### PR TITLE
Rebalances Hearty Punch

### PR DESCRIPTION
**Pull request by Repukan**
Changed healing rates to be more in par with Atropine but not as weak.
Increased metabolization rate.

balance: Rebalanced healing properties. Metabolization rate upped.

It still allows you to keep on fighting during duress and its combat properties have remained untouched. Also, I think it's an oversight that something with this much alcohol could heal what would be proper ethanol damage suffered from consuming what is a high amount of BAC you'll get from drinking a cocktail like this

### CODE REVIEW

**Comment by arsserpentarium** [comment]
why bother to make it this way?
Bartender always was a shaman.

**Comment by Repukan** [comment]
Not an argument

**Comment by Repukan** [comment]
It's not an argument because the drink itself still contains its strong combat properties without being busted.

There's no binary "Make it easier if you make it less strong or remove it" hard wall here. It is still a reference to where it came from, it mimics the strength of the reference perfectly fine, and with this it isn't a "chug and be safe from death for five minutes drink."

**Comment by kevinz000** [comment]
tfw people ruin fun things again.

**Comment by Spaceinaba** [comment]
how do you know how broken hearty punch is if you cant even pass a probation

**Comment by Repukan** [comment]
Not at argument.

**Comment by arsserpentarium** [comment]
We know, that there is no arguments for Repukan.We've seen you a lot.All is either:
dumb
hugbox
powercreep

If I'm not right, please,bring link to discussion, where you were convinced by arguments.

**Comment by Repukan** [comment]
An ad-hom is not an argument challenging the change presented in the PR.

Also the burden of proof is on you so w/e

**Comment by Memelizard** [comment]
My question is why try to nerf one of the most time consuming/tedious to make healing items in the game? It gets its power from the effort you have to put in to make a decent batch.

Not to mention its a nerf to what is already a bottom tier power level job while other service jobs (Chaplain/Botanist) already beat it in power.

**Comment by Repukan** [comment]
I'd be really iffy to calling one of the jobs with a round start gun, enough ammo to last you a skirmish against many opponents, and the ability to make stun chems without leaving their work area bottom tier.

**Comment by Repukan** [comment]
Also "Other departments are already batshit unbalanced so this should also remain batshit unbalanced" makes for poor defense because arms race powercreep only servers to dilute the overall enjoyability of other jobs.

**Comment by Memelizard** [comment]
You know as well as I do that the Bartender does not have lethal shotgun rounds roundstart.

You also know that that shotgun only has two shots before a reload is needed and both need to hit for a knockdown, not to mention being troubling when dealing with multiple foes.

The Chaplain and Botanist later ceilings far outstrip having a roundstart gun that you can only print ammo for on meta without breaking in somewhere. 

And those "stun chems" have to be directly applied via a dropper or stolen spray bottle unless you print shotgun darts. 

If you want to find something to nerf, there are plenty more worthwhile targets in science, medical, space ruins or Wizard to look at. Not nerfing a hard to make healing drink that is made maybe once every 20 rounds that is worse than a tricord pill dental implant.

This is an "I ded" PR plain and simple.

**Comment by 81Denton** [comment]
I'd rather up the metabolization rate instead so people can't preemptively chug hearty punch to get back out of crit. 
That way it's more like a superior atropine.

>the ability to make stun chems without leaving their work area
Bartenders don't have morphine available for Neurotoxin. They also don't have syringe guns, spray bottles or droppers to deliver chems.

**Comment by Plapatin** [comment]
i ded

**Comment by CosmicScientist** [comment]
>Oxy-Loss: With the extremely low metabolization rate of HP, it allows you to enter and exit crit extremely quickly. You can easily come back from crit from full buckshot at 3 tiles or so.

>Tox-Loss: Self Explanatory. It is alcohol, not a medicine. All that ethanol from the mixed drinks should hurt you, not heal you. If it heals then it's magic or someshit and it belongs on lavaland or wizards den.

If this is an alcohol and not a medicine, you should be removing brute, fire and clone healing. I understand that seeing someone shrug off a complete load of buckshot isn't fun but they're not fast are they? Can't chemists mix together a pill that's more effective and sits in a bottle with others?

If metabolisation rate is your problem then why can't you touch its metabolization_rate var to not metabolise at 10% of normal?

**Comment by 81Denton** [comment]
>Can't chemists mix together a pill that's more effective and sits in a bottle with others?

For brute/burn/oxyloss, you can get close with a mix of kelotane/bicardine/atropine/epinephrine/oxandrolone/salicyclic acid.

Epinephrine caps oxyloss, atropine/goof chems heal out of crit, yadda yadda blah blah

Quicker to mix than having to head over to the bar and convincing the bartender to give up a bunch of his bottles.

**Comment by CosmicScientist** [comment]
I guess we should remove this drink mix then - it doesn't incentivise going to the bar over chemistry.

**Comment by arsserpentarium** [comment]
No,Repukan.Balanced doesn't mean puny ,weak and without fun things.
It means, that there is fair diiculty to obtain fun things.And ways to counter them for other players.

**Comment by Repukan** [comment]
>MemeLizard

Ranged combat opens up setting on fire with alcohol/bashing and others and your large amount of ammo ensures that if you can hit at least two they won't get up quickly. You even have enough cable to cuff them.

>If this actually needs nerfing (doubt it), I'd rather up the metabolization rate instead so people can't preemptively chug hearty punch to get back out of crit.

I mean I might but I wanted some proper feedback that wasn't shitposting first.

>i ded

Can't be ded if I can't play

>If this is an alcohol and not a medicine, you should be removing brute, fire and clone healing. I understand that seeing someone shrug off a complete load of buckshot isn't fun but they're not fast are they? Can't chemists mix together a pill that's more effective and sits in a bottle with others?

If metabolisation rate is your problem then why can't you touch its metabolization_rate var to not metabolise at 10% of normal?

Side question, if you don't want someone coming back from buckshot, remove healing brute unless I don't know how this works; does this heal faster than you gain oxyloss?

Pills can't be mixed, you'd have to manually eat each one so it could kick in, which opens up storage problems. I wanted more feedback before touching metabo rate. You recover enough oxyloss to actually not fall and stay put.

Overall even in pill form it is pretty busted, and the reason I didn't touch the healing rate of fire and brute was because it is a reference to a healing drink from another game (Bastion.)

It's a catch 22 because nerfing it too much then makes it just a weak limp reference.

**Comment by CosmicScientist** [comment]
tfw didn't realise metab rate doesn't affect the chem applying its effects.

Since this is for anyone incapacitated I'd rather it just to be something you feed to someone on the floor as the bartender, with whatever drunk side effect there is.

**Comment by DaedalusGame** [comment]
>Pills can't be mixed
falsehood is not an argument

**Comment by duncathan** [comment]
but why

**Comment by Repukan** [comment]
>Pills can't be mixed
>falsehood is not an argument

Didn't know there was a way to make a pill with multiple chems in it.

**Comment by fludd12** [comment]
I've never really seen hearty punch's power abused much, especially for the incredibly low quantities it makes at a time. Were these changes spurred by a recent round or complaint, or made simply by looking at it's potential in the code?

Aside from the occasional relevant argument in the comment section above, its point is to work as a rapid crit revival method - [in fact, it's actually a reference to a video game](http://bastion.wikia.com/wiki/Hearty_Punch). My suggestion is, if you want to nerf it without having people argue forever about the changes, you can make it work a lot more like it does in Bastion, and here's how:

Have it do nothing but be a particularly strong drink, unless the person enters crit. At that point, it "activates," and begins healing with the current values. After all, the drink in bastion was designed to get the player back on their feet more times than they would normally be allowed.

Once the player is above 50% health, the drink can then "deactivate," at which point you could remove all, or maybe simply half, of the drink from them.

**Comment by Repukan** [comment]
It already works when your health drops to 0, the reason this nerf came about was because with the advent of chem macros (and apparently now I was informed we have bar macros) this could be extremely abused.

It is akin to fighting someone with old viro, and given that at least one role starts with it (detective gets a full flask with 30u on it roundstart) makes fights against an invisible threat that didn't even give enough of a tell due to its comparably low boozepower.

I'm aware of the Bastion reference which is why I wanted to nerf it instead of opening a removal PR. Making it metabolize faster, reduce its healing rate, and increasing its boozepower will make it more of a risky but still useful combat drink rather than a "drink to cheat death for an insanely long time."

**Comment by Repukan** [comment]
I should add, it only works when your health hits 0

**Comment by fludd12** [comment]
Ah, well, fair enough I suppose. Perhaps there is some other way to nerf it, though? Such as, when you hit zero, applying a knockback effect to people surrounding you, then rapidly healing you and depleting at a faster rate until either you are 50% healed, or it runs out? This gives a counterplay, and makes it significantly closer to its effect in Bastion, but simultaneously still accomplishes the goal of the player surviving longer, and potentially recovering enough to get back on their feet and back into the fight.

**Comment by fludd12** [comment]
Ah - to clarify, it would also keep you on the ground for that time, as well. Think of it like a revival coma, but you're still conscious.

**Comment by Repukan** [comment]
I'm not that good at coding tbh

**Comment by fludd12** [comment]
Neither am I, but I'll give it a shot. Keep this PR open, though! If you think it needs a nerf, I won't argue. I am just like a weird pinata full of ideas that are hard to make.

**Comment by DaedalusGame** [comment]
all this currently does defies the whole point of being critical for this to work, because it'll be gone in 30 cycles after imbiding. additionally you got rid of two damage types that while it makes sense for alcohol not to heal them, but they're rare in terms of actual combat or anything worth doing really.
keep everything as is but bump boozepwr to 90
the whole point of alcohol is that it deals liver damage per reagent cycle, 30u (det flask) stays in your body for 750 cycles, the formula for taking liver damage is sqrt(volume) * boozepwr * 0.005 per tick. I graphed it and with your current changes you get the full heal (this heals more than twice as much as atropine for the two important damage types) and a total of less than 10 liver damage (100 is needed for liver failure) for 30u, 60u barely does more than 20 damage to your liver.
Liver damage heals slowly (0.1 per cycle) and I'm pretty sure it doesn't go away from chem usage or any other way but I could be wrong, if you take 6 units at the current metabolization speed it will last 150 cycles, with 90 boozepwr (and the -0.1 damage factored in) you will take about 60 liver damage. You would reach liver failure at cycle 190 thereabouts (probably somewhere between 3 minutes and 6 minutes). Liver failure deals 8 tox damage per cycle *and prevents chems (like hearty punch) from metabolizing* so you don't immediently heal it back either.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -1108,20 +1108,18 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	id = "hearty_punch"
 	description = "Brave bull/syndicate bomb/absinthe mixture resulting in an energizing beverage. Mild alcohol content."
 	color = rgb(140, 0, 0)
-	boozepwr = 10
-	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	boozepwr = 90
+	metabolization_rate = 1.0 * REAGENTS_METABOLISM
```
Comment (CosmicScientist) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: 1*x = x
You can just remove multiplying it by 1.0 and it'll be the same calculation as with it.

**Comment by KorPhaeron** [comment]
The detective starts with a full flask of this so getting it isnt difficult

It consumes .04 units per tick so it doesnt take much of it to last most of the round so it being tedious to make still wouldnt be a big issue

Once its in your system you are mostly immune to anything but stun+cuffs

I have been accused of being a ling as detective before

**Comment by KorPhaeron** [comment]
Bords idea sounds pretty good

**Comment by Repukan** [comment]
Will modify the pr then

**Comment by Repukan** [comment]
I still insist however, in that it shouldn't heal oxy and tox damage

**Comment by arsserpentarium** [comment]
Why do you insist on changes in game, you can't play, which are related to mechanics, which you didn't used even when you played?

**Comment by Repukan** [comment]
Not an argument.

**Comment by ExcessiveUseOfCobblestone** [comment]
now why would you put a better atropine with no OD threshold or any practical downsides in an entire flask roundstart...

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: if you keep metab as is and tenfold raise the boozepwr that should heck over your liver for sure no?

**Comment by Repukan** [comment]
I'll remove it coach dw

also @KorPhaeron I've restored the healing effects it had as per what you said but modified them to be only slightly stronger than Atrophine. I still feel it should axe the Oxy and Tox loss but it's your call tbh.

**Comment by arsserpentarium** [comment]
Yes,it's a question. Why you even need to do it?

**Comment by Repukan** [comment]
Because it's unbalanced?

Are you too ESL to come up with a better argument against this PR that isn't "You're permabanned?"

**Comment by ExcessiveUseOfCobblestone** [comment]
I mean if it wasn't maxed out in a roundstart flask then the argument could be it's a pain to make in bulk.

**Comment by arsserpentarium** [comment]
Not an argument.

1. Who is suffering from this balance flaw?
2. Who is overpowered this way?
3. How often use of this feature is occured?

**Comment by Repukan** [comment]
1. Any player that must engage in combat because until this was a free get out of death card with no good tells (boozepower was pitifully low and didn't even give slurr sometimes)

2. Bartenders, Detectives and anyone with bar access really thanks to macros

3. Often enough to be noticed by the change maker.

**Comment by ExcessiveUseOfCobblestone** [comment]
I actually used to put this in cryo when I played chemist with antihol ;)

**Comment by Repukan** [comment]
You know, I just realized I don't owe you explanations.

**Comment by arsserpentarium** [comment]
Maybe because otherwise you owe explanaitions, how are you avoiding ban?

**Comment by Repukan** [comment]
>Thinks i'm ban evading.

lmao

**Comment by ExcessiveUseOfCobblestone** [comment]
Imagine defending a better form of cryo present roundstart without the sleep/slow or coldreq.

**Comment by arsserpentarium** [comment]
@Repukan Yes, because otherwise you can't notice it.
@ExcessiveUseOfCobblestone but  very limited orm of cryo.
anyway you can barely roundstart breed shitton of slimes and make pyroxadone.

**Comment by ExcessiveUseOfCobblestone** [comment]
it's not limited wtf the detective gets it roundstart and I could make 6u easily x however much I felt like making, and that was before macros (which can be done to make hearty punch minus soda)

**Comment by epochayur** [comment]
remember, if someone says something you don't like, just reply with "not an argument"

**Comment by Repukan** [comment]
Attacking the person and not the issue it's not an argument, but a logical fallacy.

**Comment by ExcessiveUseOfCobblestone** [comment]
MRTY was banned and for a bit actively evaded and his PRs did not get closed.

Repukan has not evaded to our (my) knowledge at all so I don't think his status of being banned should even be brought up in any sentence not ending in :^) on this or any pr

**Comment by PKPenguin321** [comment]
Unnecessary

**Review by Cyberboss — APPROVED** [review]

**Comment by 81Denton** [comment]
With this, hearty punch would be stronger than pure atropine, but inferior to any halfassed atropine+epinephrine+goofchem mix. 
While retaining the whole 1/15 mix amount and having boozepwr close to hooch.
I bet 3 GBP that this will be merged before Friday

**Comment by AlexTheSysop42** [comment]
>You know, I just realized I don't owe you explanations.

you kinda do if you want your PRs merged

**Comment by Repukan** [comment]
@81Denton

I could lower its values to be slightly less than Atropine.

@Alex 
Only to mantainers

**Comment by AlexTheSysop42** [comment]
And maintainers are the only people whose questions mean anything?

**Comment by Repukan** [comment]
No, player feedback is important too, but actual feedback, not accusations of ban evading with no water to hold.

**Comment by AlexTheSysop42** [comment]
They didn't mention evasion until after your comment about not needing to answer.

**Comment by AlexTheSysop42** [comment]
Sorry, I was mistaken. You made that reply in response to something that didn't talk about evasion, but it had been brought up before.

**Comment by AlexTheSysop42** [comment]
Also, the macro thing could easily be solved just by making it require external chems or heating. (Maybe also change the det's flask to that sec drink that heals less.)

**Comment by Repukan** [comment]
I don't mean to be a hypocrite but I've entertained the other guy far past my patience limit.

I do intend to change the dets flask to whiskey, since a few noir movies and such have them drinking whiskey and I feel it sticks more to the noir theme, if that makes sense. I'll look into Cuba libre code to see if I can make detectives get a small healing boost from whiskey if they don't already, which would go hand in hand.

**Comment by ExcessiveUseOfCobblestone** [comment]
It does require an external chem.

I mean it's possible to just not have hearty punch heal that much and save a lot of trouble of playing contributor hopscotch.

**Comment by AlexTheSysop42** [comment]
@Repukan there's already like the triple sec or quadruple sec or something

@ExcessiveUseOfCobblestone didn't know, assumed it didnt since people were talking about macro exploits

**Comment by yorii** [comment]
Why is this change necessary? It will still heal you to health > 0. And you should also change the description if you intend to change the alcohol content.

**Comment by yorii** [comment]
Actually this PR will buff hearty punch quite a bit.

**Comment by Xhuis** [comment]
In which repukan continues to change things in a game he isn't allowed to play.

**Comment by Repukan** [comment]
Shouldn't you be making your gamemode not an absolute shit to play instead of shitposting in PRs?

**Comment by Xhuis** [comment]
Shouldn't you stop pretending to be relevant?

**Comment by optimumtact** [comment]
actually repukan just suggested the change.

if you think about it, I made the change

**Comment by Xhuis** [comment]
The difference is that you actually contribute something to the server in your spare time, oranges.

**Comment by Repukan** [comment]
I'm only as relevant as you make me, Xhuis.

Perhaps you'll realize this after you break down again from people not happily wagging their tails to your beck and design that falls flat every time.

Maybe you should give Clockcult back to Joan. At least they were better at working on the mode and fixing exploits and bugs than you will ever be.

**Pull request merged**